src/account/sync/mod.rs

```diff
@@ -998,6 +999,19 @@ async fn perform_sync(
         };
     }
 
+    // If we discover the account and the first public address isn't added, we will do it here
+    if return_all_addresses && !addresses_to_save.iter().any(|a| *a.key_index() == 0 && !a.internal()) {
+        log::debug!("[SYNC] adding first public address because we're discovering this account");
+        addresses_to_save.push(
+            (*public_addresses
+                .collect::<Vec<&Address>>()
```

Comment:
Do we really need to collect here ? Can't we juste `next` the iterator ?

Reply:
No, changed